### PR TITLE
Support dart plugins in test command

### PR DESCRIPTION
**Pull request by HakkyuKim**
Allows injecting dart plugins for `flutter-tizen test integration_test`. It's sort a hack, I'm open to better suggestions.

Contributes to #168.

You can test with plugins that require dart code: `package_info_plus`, `connectivity_plus`, ...

### CODE REVIEW

**Comment by HakkyuKim** [comment]
On second thought, I don't think it's a good idea to change the code, users will be able to see their code disappearing in IDE. I can add some comments in the "replacing" code, but still it's not a good solution. I'll try again.

**Comment by swift-kim** [comment]
You can either

- pass fake `testFiles` to `test_core` by providing a custom `testWrapper`,
- create a fake entrypoint in `TizenDevice.startApp()` by checking if `mainPath` contains `flutter_test_listener.`, or
- [refactor the current plugin architecture of flutter-tizen and create a build Target which does a similar thing to the upstream's `DartPluginRegistrantTarget`](https://github.com/flutter-tizen/plugins/issues/123#issuecomment-875580942). This is the most preferable way but is a little bit complicated. I'll do some PoC implementation if you agree.

**Comment by HakkyuKim** [comment]
@swift-kim 
Wow, Thank you for the solutions. I haven't checked the last one yet, if it's the most preferable way, then yes I would appreciate the help! If it's something that you're not actively focusing on at the moment, I think the first option looks good as a temporary solution. Please see if the updated commit is what you meant.

File: lib/commands/test.dart [hunk 1]
```diff
@@ -4,13 +4,118 @@
...
 // @dart = 2.8
 
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/cache.dart';
 import 'package:flutter_tools/src/commands/test.dart';
+import 'package:flutter_tools/src/test/runner.dart';
+import 'package:flutter_tools/src/test/test_wrapper.dart';
+import 'package:flutter_tools/src/dart/language_version.dart';
```
Comment (swift-kim) [inline lib/commands/test.dart:14]: Sort the imports alphabetically.

File: lib/commands/test.dart [hunk 2]
```diff
@@ -4,13 +4,118 @@
...
+      final FlutterProject flutterProject = FlutterProject.current();
+      final PackageConfig packageConfig = await loadPackageConfigWithLogging(
+        flutterProject.directory.childFile('.packages'),
+        logger: globals.logger,
+      );
+      final File pluginRegistrant = TizenProject.fromFlutter(flutterProject)
+          .managedDirectory
+          .childFile('generated_plugin_registrant.dart');
+
+      runnerDir = globals.fs.systemTempDirectory..createTempSync();
```
Comment (swift-kim) [inline lib/commands/test.dart:80]: Did you mean
```suggestion
      runnerDir = globals.fs.systemTempDirectory.createTempSync();
```

File: lib/commands/test.dart [hunk 3]
```diff
@@ -4,13 +4,118 @@
...
+  entrypoint.main();
+}
+''');
+        newArgs.add(runnerFile.absolute.path.toString());
+      }
+      await test.main(newArgs);
+    } finally {
+      if (runnerDir.existsSync()) {
+        runnerDir.deleteSync(recursive: true);
+      }
```
Comment (swift-kim) [inline lib/commands/test.dart:111]: You don't have to delete temp directories manually.

File: lib/commands/test.dart [hunk 4]
```diff
@@ -4,13 +4,118 @@
...
+
+import '${testFile.absolute.path.toString()}' as entrypoint;
+import '${pluginRegistrant.absolute.path.toString()}';
+
+Future<void> main() async {
+  registerPlugins();
+  entrypoint.main();
+}
+''');
+        newArgs.add(runnerFile.absolute.path.toString());
```
Comment (swift-kim) [inline lib/commands/test.dart:105]: Redundant `toString()`'s.

**Review by swift-kim — COMMENTED** [review]
Looks okay for now. I'm going to implement the third solution soon so you can use this workaround until then.

File: lib/commands/test.dart (hunk 4 above)
Comment (swift-kim) [reply lib/commands/test.dart]: Line 105 :)

**Review by swift-kim — APPROVED** [review]

**Pull request merged**